### PR TITLE
openapi: add description for PciHostDevice properties

### PR DESCRIPTION
**Pull request by dankenigsberg**
Make https://kubevirt.io/api-reference/master/definitions.html#_v1_pcihostdevice easier to understand.

Signed-off-by: Dan Kenigsberg <danken@redhat.com>

@vladikr is this the place to add this content?

```release-note
NONE
```

### CODE REVIEW

**Comment by EdDev** [comment]
> @vladikr is this the place to add this content?

Sorry for barging in. I think you need to add it on the type itself: https://github.com/kubevirt/kubevirt/blob/7c3b33528ac61ec38e8ae5d54bb025988464e1ea/staging/src/kubevirt.io/client-go/api/v1/types.go#L1915

Then it will get generated all over.

**Comment by EdDev** [comment]
@dankenigsberg , I think you need to run `make generate` (and add the artifacts to the commit) before you push.

**Comment by dankenigsberg** [comment]
/retest

https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/5882/pull-kubevirt-unit-test/1406899263861952512 fails due to https://github.com/kubevirt/kubevirt/issues/5753

File: api/openapi-spec/swagger.json [hunk 1]
```diff
@@ -10809,12 +10809,14 @@
...
     "properties": {
      "externalResourceProvider": {
+      "description": "The name of the resource that is representing the device. Exposed by a device plugin and requested by VMs. Typically of the form vendor.com/product_nameThe name of the resource that is representing the device. Exposed by a device plugin and requested by VMs. Typically of the form vendor.com/product_name",
       "type": "boolean"
      },
      "pciVendorSelector": {
       "type": "string"
      },
      "resourceName": {
+      "description": "\"The vendor_id:product_id tupple of the PCI device",
```
Comment (fgimenez) [inline api/openapi-spec/swagger.json:10819]: Why do you need the `\"`? Not sure if it is breaking things.
```suggestion
      "description": "The vendor_id:product_id tuple of the PCI device",
```

File: manifests/generated/kv-resource.yaml [hunk 2]
```diff
@@ -2094,10 +2102,18 @@ spec:
...
+                                by VMs. Typically of the form vendor.com/product_nameThe
+                                name of the resource that is representing the device.
+                                Exposed by a device plugin and requested by VMs. Typically
+                                of the form vendor.com/product_name
                               type: boolean
                             pciVendorSelector:
                               type: string
                             resourceName:
+                              description: "The vendor_id:product_id tupple of the
+                                PCI device'
```
Comment (fgimenez) [inline manifests/generated/kv-resource.yaml:2116]:
```suggestion
                              description: The vendor_id:product_id tuple of the
                                PCI device
```

File: pkg/virt-operator/resource/generate/components/validations_generated.go [hunk 3]
```diff
@@ -638,10 +638,18 @@ var CRDsValidation map[string]string = map[string]string{
...
+                          VMs. Typically of the form vendor.com/product_nameThe name
+                          of the resource that is representing the device. Exposed
+                          by a device plugin and requested by VMs. Typically of the
+                          form vendor.com/product_name
                         type: boolean
                       pciVendorSelector:
                         type: string
                       resourceName:
+                        description: '"The vendor_id:product_id tupple of the PCI
+                          device'
```
Comment (fgimenez) [inline pkg/virt-operator/resource/generate/components/validations_generated.go:652]:
```suggestion
                        description: 'The vendor_id:product_id tuple of the PCI
                          device
```

**Review by fgimenez — COMMENTED** [review]
Looks like something is wrong with the generated kubevirt-operator.yaml https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/5882/pull-kubevirt-e2e-k8s-1.19-sig-compute/1407262002291675137#1:build-log.txt%3A4369
`error parsing /home/prow/go/src/github.com/kubevirt/kubevirt/_out/manifests/release/kubevirt-operator.yaml: error converting YAML to JSON: yaml: line 354: did not find expected key`

File: staging/src/kubevirt.io/client-go/api/v1/types.go [hunk 4]
```diff
@@ -1910,9 +1910,17 @@ type PermittedHostDevices struct {
 // PciHostDevice represents a host PCI device allowed for passthrough
 // +k8s:openapi-gen=true
 type PciHostDevice struct {
-	PCIVendorSelector        string `json:"pciVendorSelector"`
-	ResourceName             string `json:"resourceName"`
-	ExternalResourceProvider bool   `json:"externalResourceProvider,omitempty"`
+	PCIVendorSelector string `json:"pciVendorSelector"`
+	// "The vendor_id:product_id tupple of the PCI device
```
Comment (rmohr) [inline staging/src/kubevirt.io/client-go/api/v1/types.go:1914]: This `"` seems to be causing the unwanted character in the docs.

Comment (dankenigsberg) [reply staging/src/kubevirt.io/client-go/api/v1/types.go]: Thanks. I took the lazy path of dropping the quote mark over fixing whatever is broken in `make generate`

File: staging/src/kubevirt.io/client-go/api/v1/types.go [hunk 5]
```diff
@@ -1910,9 +1910,17 @@ type PermittedHostDevices struct {
 // PciHostDevice represents a host PCI device allowed for passthrough
 // +k8s:openapi-gen=true
 type PciHostDevice struct {
-	PCIVendorSelector        string `json:"pciVendorSelector"`
-	ResourceName             string `json:"resourceName"`
-	ExternalResourceProvider bool   `json:"externalResourceProvider,omitempty"`
+	PCIVendorSelector string `json:"pciVendorSelector"`
+	// The vendor_id:product_id tupple of the PCI device
```
Comment (rmohr) [inline staging/src/kubevirt.io/client-go/api/v1/types.go:1914]: I think you have put by incident the comment for the fields below the fields than above them.

**Review by rmohr — CHANGES_REQUESTED** [review]

File: staging/src/kubevirt.io/client-go/api/v1/types.go (hunk 5 above)
Comment (dankenigsberg) [reply staging/src/kubevirt.io/client-go/api/v1/types.go:1914]: oh dear. thanks for the babysitting.

**Pull request merged**